core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/compare/MinHandler.java

```diff
@@ -35,6 +36,7 @@
  */
 public class MinHandler implements OutputOperationHandler<Min, Element> {
     @Override
+    @SuppressWarnings("PMD.UseTryWithResources")
```

Comment:
Is it worth having this rule? What percentage are being suppressed?

Reply:
This has to be suppressed because while Operations are closable, they can't be used in a try with resources because in Java 8 they must be defined inside the try. They also can throw exceptions and the way this is handled with `CloseableUtil.close` (it ignores the exceptions) is incompatible with how Java expects this to work.
If we refactored to remove closableUtil and used Java 9+, this is how it could be done:
```
try (operation) {
    return getMin(operation.getInput(), operation);
} catch (IOException eWhichUsedToBeIgnoredByClosableUtil){
    // Do error handling
}
```

Reply:
I just noticed lots of suppressing this rule so wasn't sure if it was worth keeping in the main rule set

Reply:
I think it's worth keeping in as it's only operations where we can't use it properly and they have now all been suppressed. For opening files it works fine. In future, once we've moved to Java 11, we can come back and fix these suppressions, as shown above. Is there an issue about Java 11 or was that closed? It would be good to track all the things which can be improved once it's used.

Reply:
Yeah okay that sounds good. I don't think there is, perhaps there should be like you said to track all the things that could be done

Reply:
I've created [this issue 2874](https://github.com/gchq/Gaffer/issues/2874) to cover this idea. It links back to this thread.